app/analysis/edit/controller.js

```diff
@@ -0,0 +1,153 @@
+import Controller from '@ember/controller'
+import { inject as service } from '@ember/service'
+import { task } from 'ember-concurrency'
+import { AnalysisQueryParams } from '../index/controller'
+import { get } from '@ember/object'
+import { dasherize, underscore } from '@ember/string'
+import { cleanParams } from 'timed/utils/url'
+import computed from 'ember-computed-decorators'
+import { inject as controller } from '@ember/controller'
+
+/* eslint-disable ember/avoid-leaking-state-in-ember-objects */
+export const AnalysisEditQueryParams = AnalysisQueryParams.extend({
+  id: {
+    defaultValue: [],
+    replace: true,
+    refresh: true,
+    serialize(arr) {
+      return (arr && arr.join(',')) || ''
+    },
+    deserialize(str) {
+      return (str && str.split(',')) || []
+    }
+  }
+})
+/* eslint-enable ember/avoid-leaking-state-in-ember-objects */
+
+const prepareParams = params => {
+  return cleanParams(
+    Object.keys(params).reduce((parsed, key) => {
+      let serialize = get(
+        AnalysisEditQueryParams,
+        `queryParams.${key}.serialize`
+      )
+      let value = get(params, key)
+
+      return key === 'type'
+        ? parsed
+        : {
+            ...parsed,
+            [underscore(key)]: serialize ? serialize(value) : value
+          }
+    }, {})
```

Comment:
This seems to be pretty much that same as in the statistics controller. Since the method is pretty high-level it might be worth to

* extract it
* add a docstring

Reply:
Done!